.github/workflows/tests.yml

```diff
@@ -45,12 +44,18 @@ jobs:
         flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
     - name: Test without pydantic
       run: |
-        pytest
+        pytest --cov=tap
     - name: Test with pydantic v1
       run: |
         python -m pip install "pydantic < 2"
-        pytest
+        pytest --cov=tap --cov-append
     - name: Test with pydantic v2
       run: |
         python -m pip install "pydantic >= 2"
-        pytest
+        pytest --cov=tap --cov-append
+
+    - name: Upload coverage reports to Codecov
```

Comment:
FYI new behavior (evidently) runs a code coverage report for PRs to main. Current behavior only runs a coverage report on pushes to main

Reply:
Maybe [there's a way](https://stackoverflow.com/a/64222104/18758987) to upload the final coverage report after the matrix is done. Lmk if that behavior is more desirable than the current one, which uploads a coverage report per matrix entry (15 of them)